tests/Suggestor/Suggestor.test.ts

```diff
@@ -150,3 +150,75 @@ describe.each([
         markdownTable.verify();
     });
 });
+
+describe('onlySuggestIfBracketOpen', () => {
+    const buildSuggestions: SuggestionBuilder = onlySuggestIfBracketOpen(
+        // A dummy SuggestionBuilder. Just used to validate that results are returned, results aren't meaningful
+        () => [{} as unknown as SuggestInfo],
+        [
+            ['(', ')'],
+            ['[', ']'],
+        ],
+    );
+
+    const emptySuggestion = [] as unknown;
+
+    /**
+     * Given a string with **exactly** one vertical bar (`|`), returns the string with the vertical bar removed
+     * and the index of the bar.
+     *
+     * This is used as a helper when writing tests, since it provides a less error prone
+     * and more readable way to represent a cursor's position, denoted by the vertical bar, and the string.
+     *
+     * @param line - A string that contains exactly one vertical bar (`|`)
+     * @returns A tuple of the line without the vertical bar, and the index of the vertical bar.
+     */
+    function cursorPosition(line: string): [lineWithoutCursor: string, cursorIndex: number] {
+        const line_without_cursor = line.replace('|', '');
+        // Check that the cursor marker appears exactly once in each input string:
+        expect(line_without_cursor.length).toEqual(line.length - 1);
```

Comment:
I'll go ahead and merge, and then fix this on default...

But just to point out, it's always worth "testing" the tests.. I put in some extra `|` lines, and the tests still passed.

It turns out that 177 needs to be the following for multiple instances to be replaced:

```diff
-        const line_without_cursor = line.replace('|', '');
+        const line_without_cursor = line.replace(/\|/g, '');
```

It needs both the regex, and the `g` option.